### PR TITLE
Fix for copy/paste buttons in appearance and effects panels

### PR DESCRIPTION
**Pull request by shaoshing**
This PR makes two changes to the copy/paste buttons:

1 copy/paste buttons in the appearances now include opacity, radius, and blend mode. This should apply for text layer, vector layer, and other layers (e.g. SO). You can checkout the video for more details of the bahaviors: https://www.dropbox.com/s/lao6v0xo6nezt0g/copy-styles.mov?dl=0. Addresses #2755 

2 copy/paste buttons in the effects panel should only apply to layer effects (used to include appearance styles). Same for the appearance panel, buttons should not copy/paste layer effects. 

###### Known issues that are not introduced from this PR (I will create separate issues later)

1 copy/paste appearance between two text layers with different opacity result in number change back and forth.

text-styles

2 undo paste-styles some times does not reset the appeanrace panel

### CODE REVIEW

**Comment by shaoshing** [comment]
@chadrolfs or @ktaki, when you have time, can you give this branch a try? I am not confident enough that I've tested all dark corners :P

File: src/js/actions/layers.js [hunk 1]
```diff
@@ -1544,19 +1544,22 @@ define(function (require, exports) {
...
+     * @param {boolean=} options.coalesce Whether to coalesce this operation's history state
      * @return {Promise}
      */
-    var setOpacity = function (document, layers, opacity, coalesce) {
+    var setOpacity = function (document, layers, opacity, options) {
         layers = layers.filterNot(function (layer) {
             return layer.isBackground;
         });
+        
+        options = _.merge({ coalesce: false }, options);
```
Comment (iwehrman) [inline src/js/actions/layers.js]: I guess it doesn't really matter, but why is this line necessary? If you're doing `!!options.coalesce` below, this seems like a no-op.

Comment (iwehrman) [inline src/js/actions/layers.js]: Oh, maybe because you want a fresh object reference below. It still seems like the first param could be an empty object literal, but again it probably doesn't really matter.

**Comment by iwehrman** [comment]
I do not see anything objectionable here! If @chadrolfs or @ktaki can do some functional testing, I'm happy to merge.

**Comment by iwehrman** [comment]
Tentatively assigning to @chadrolfs for this, but feel free to bounce it elsewhere if that doesn't work.

**Comment by chadrolfs** [comment]
Logged, #3049 and #3050

**Comment by chadrolfs** [comment]
Looks like there is a bug similar to #3016 

shape A has (CC Library "layer effects": effects - a lot), shape B has one drop shadow
copy Effects from B, select A, paste
result, 2 drop shadows

**Comment by chadrolfs** [comment]
^^ is fixed by PR 3041.  Merging.
